cdap-test/src/main/java/co/cask/cdap/test/TestManager.java

```diff
@@ -45,6 +50,37 @@ ApplicationManager deployApplication(Id.Namespace namespace,
                                        Class<? extends Application> applicationClz, File... bundleEmbeddedJars);
 
   /**
+   * Deploys an {@link ApplicationTemplate}. Only supported in unit tests.
```

Comment:
Curious about the comment - "only supported in unit tests". Where else could this could be used?